packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -408,20 +415,24 @@ export class AttributionCollection implements IAttributionCollection<Attribution
 	): void {
 		const { channels } = summary;
 		assert(
+			// Destructuring here would require renaming the variables, since seqs is declared below
+			// eslint-disable-next-line unicorn/consistent-destructuring
 			summary.seqs.length === summary.posBreakpoints.length,
 			0x445 /* Invalid attribution summary blob provided */,
 		);
 
 		const extractOntoSegments = (
 			{ seqs, posBreakpoints }: SequenceOffsets,
 			assignToSegment: (collection: AttributionCollection, segment: ISegment) => void,
-		) => {
+		): void => {
 			let curIndex = 0;
 			let cumulativeSegPos = 0;
 
 			for (const segment of segments) {
 				const attribution = new AttributionCollection(segment.cachedLength);
-				const pushEntry = (offset: number, seq: AttributionKey | number | null) => {
+				// This function is defined here to allow for the creation of a new collection for each segment.
+				// eslint-disable-next-line unicorn/consistent-function-scoping
```

Comment:
@tylerbutler I find myself disabling this rule a lot. I'm not convinced it's a valuable one. How would you feel about disabling this globally?

Reply:
The docs make this sound helpful for perf, at least in V8: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md

Reply:
Speaking just for myself, this is one of the "dumb mistakes" I often make, especially when writing new code or refactoring.

Reply:
I wasn't aware of the performance implications, but I definitely disagree with the "improves readability" argument they make. Placing functions closer to their callsite(s) makes the code much easier to reason about in cases like this one IMO.

Reply:
But if there are performance benefits, then this may be worth keeping. And I believe we've already disabled this for tests.